ToDoApp/ToDoApp/ToDoListViewModel.swift

```diff
@@ -39,9 +39,18 @@ final class ToDoListViewModel: ObservableObject {
         }
         fetchToDoTasks()
     }
+    // Tap Edit button
+    func editButtonTapped(todo: TaskModel) {
+        let createTaskViewModel: CreateTaskViewModel = CreateTaskViewModel(taskService: taskService, mode: .editExistingTask(todo)) { [weak self] in self?.resetDestination()
+        } onSaved: { [weak self] in
+            self?.resetDestination()
+            self?.fetchToDoTasks()
+        }
+        destination = .addTask(createTaskViewModel)
+    }
     // Tap add button
     func addButtonTapped() {
```

Comment:
it seems that editButtonTapped and addButtonTapped are doing pretty much the same logic except passing different mode values to the CreateTaskViewModel init. can we extract implementation to a private method and pass a mode value inside?

Reply:
I think yes, let me try